service/collector/vault.go

```diff
@@ -56,8 +57,15 @@ func NewVault(config VaultConfig) (*Vault, error) {
 }
 
 func (v *Vault) Collect(ch chan<- prometheus.Metric) error {
+	ctx := context.Background()
+
 	secret, err := v.vaultClient.Auth().Token().LookupSelf()
-	if err != nil {
+	if IsVaultAccess(err) {
+		v.logger.LogCtx(ctx, "level", "debug", "message", "vault not reachable")
```

Comment:
Should we add some explanation why vault is not reachable in the log? like `vault not reachable due to upgrade in progress`?

Reply:
Additional information would make sense. I am just not sure what to go for. The error and the assumption are different things. What about adding another log line `vault upgrade in progress`?

Reply:
I prefer single log because of my personal modus operandi. I'm pretty sure I could easily miss the other log. But do what you think it's right. I approved already.

Reply:
DOes it make sense here to collect a metric like did in other case https://github.com/giantswarm/cluster-service/commit/135da30bc9f30b6dc2831f9e669620a140d09ef4

Reply:
I am going with separate logs as they are separate already. We cannot simply look at a single log line anyway. 

Good shout on the additional metrics. In this case though I think that we are fine because there is already an alert when Vault is sealed and not available. The logging is only indicative for the cancelation behaviour. I think we are fine here for now. When there will be a stronger reason to add more error metrics we can gladly do this.